packages/compiler/src/render3/r3_ast.ts

```
@@ -16,6 +16,26 @@ export interface Node {
   visit<Result>(visitor: Visitor<Result>): Result;
 }
 
+/**
+ * This is an R3 `Node`-like wrapper for a raw `html.Comment` node. We do not currently
+ * require the implementation of a visitor for Comments as they only collected at
+ * the top-level of the R3 AST, and only if `Render3ParseOptions['collectCommentNodes']`
+ * is true.
+ */
+
+// IMPLEMENTATION 1) Implements Node but with a stubbed visit()
+export class Comment implements Node {
+  constructor(public value: string, public sourceSpan: ParseSourceSpan) {}
+  visit<Result>(visitor: Visitor<Result>): Result {
+    return null as unknown as Result;
```

I would go with this implementation but instead of returning `null` I would just throw a "not implemented" error.

---

Thanks for the update. I agree with Pete, implementation #1 is better because it explicitly implements the `Node` interface.